wpilibc/src/main/native/cpp/event/BooleanEvent.cpp

```diff
@@ -29,9 +33,13 @@ BooleanEvent BooleanEvent::operator!() {
   return BooleanEvent(this->m_loop, [lhs = m_condition] { return !lhs(); });
 }
 
-BooleanEvent BooleanEvent::operator&&(std::function<bool()> rhs) {
+BooleanEvent BooleanEvent::operator&&(BooleanEvent rhs) {
+  if (m_loop != rhs.m_loop) {
+    m_loop->Bind(
+        [&rhs] { rhs.m_state = std::make_shared<bool>(rhs.m_condition()); });
+  }
```

Comment:
I think this will cause trouble.

Reply:
It does. How do I check for reference equality in C++? The reason for this is because if two BooleanEvents are composed with different EventLoops, the 2nd `BooleanEvent` needs to be updated from the `EventLoop` that's actually used.